core/src/test/scala/org/apache/spark/CacheManagerSuite.scala

```diff
@@ -82,6 +84,33 @@ class CacheManagerSuite extends FunSuite with LocalSparkContext with BeforeAndAf
     assert(value.toList === List(5, 6, 7))
   }
 
+  test("cache remotely block") {
+    val blockInfo = new TimeStampedHashMap[BlockId, Boolean]
+    val result = new BlockResult(Array(5, 6, 7).iterator, DataReadMethod.Memory, 12)
+    // mock remotely received block
+    when(blockManager.get(RDDBlockId(0, 0))).thenReturn(Some(result))
+    when(blockManager.containsBlockId(RDDBlockId(0, 0)))
+      .thenReturn(false)
+    when(blockManager.putArray(RDDBlockId(0, 0), Array(5, 6, 7), StorageLevel.MEMORY_ONLY))
+      .thenThrow(new RuntimeException("putArray"))
+
+    assert(blockInfo.contains(RDDBlockId(0, 0)) === false)
+
+    val context = new TaskContextImpl(0, 0, 0, 0)
+    try {
+      val computeValue =
+        cacheManager.getOrCompute(rdd, split, context, StorageLevel.MEMORY_ONLY, cacheRemote = true)
+    } catch {
+      case e: RuntimeException =>
+        if (e.getMessage().contains("putArray")) {
+          blockInfo.putIfAbsent(RDDBlockId(0, 0), true)
+        }
+    }
+
+    // remotely block is cached now
+    assert(blockInfo.contains(RDDBlockId(0, 0)) === true)
```

Comment:
In all projects I have been part of, you would avoid `== true`, and if I poke around in some other tests, that seems to be the norm in spark as well. 
I will remove it, unless you think it might be a problem, that developers will overlook `!` in `assert(!blockInfo.contains(RDDBlockId(0, 0)))`. But its up to you :)